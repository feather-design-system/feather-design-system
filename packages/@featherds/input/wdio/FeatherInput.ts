export class FeatherInput {
  private inputSelector: string;
  get input() {
    return $(this.inputSelector);
  }
  constructor(_input: string) {
    this.inputSelector = _input;
  }

  async setValue(text: string) {
    const input = await this.input;
    await input.waitForDisplayed();
    return input.setValue(text);
  }

  async getValue() {
    const input = await this.input;
    await input.waitForDisplayed();
    return input.getValue();
  }

  async clearValue() {
    const input = await this.input;
    await input.waitForDisplayed();

    const inputValue = await input.getValue();
    const backSpaces = new Array(inputValue.length).fill("Backspace");
    await input.setValue(backSpaces);
  }
}
