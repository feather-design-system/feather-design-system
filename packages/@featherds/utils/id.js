const getSafeId = function (str) {
  str = str || "feather";
  const random = Math.floor(Math.random() * 1000000000);
  return [str.replace(/\s+/g, "-"), Date.now(), random].join("-");
};

export { getSafeId };
