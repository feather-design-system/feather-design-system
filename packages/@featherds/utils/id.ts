const getSafeId = function (str: string) {
  str = str || "feather";
  const random = Math.floor(Math.random() * 1000000000);
  return [str.replace(/\s+/g, "-"), Date.now(), random].join("-");
};

export { getSafeId };
