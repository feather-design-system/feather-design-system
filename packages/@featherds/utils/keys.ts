const KEYCODES = {
  PAGEUP: 33,
  PAGEDOWN: 34,
  UP: 38,
  DOWN: 40,
  LEFT: 37,
  RIGHT: 39,
  HOME: 36,
  END: 35,
  ENTER: 13,
  SPACE: 32,
  ESCAPE: 27,
  BACKSPACE: 8,
  DELETE: 46,
};

export { KEYCODES };
