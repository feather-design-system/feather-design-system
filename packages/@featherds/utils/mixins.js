var translateKey = {
  props: {
    translationKey: {
      type: String,
      default: "",
    },
  },
  methods: {
    translate: function (key) {
      return this.$t
        ? this.$t(this.translationKey + key)
        : "Vue i18n is not loaded";
    },
  },
};
export { translateKey };
