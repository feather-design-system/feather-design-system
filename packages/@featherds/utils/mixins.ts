const translateKey = {
  props: {
    translationKey: {
      type: String,
      default: "",
    },
  },
  methods: {
    translate: function (this: IComponent, key: string) {
      return this.$t
        ? this.$t(this.translationKey + key)
        : "Vue i18n is not loaded";
    },
  },
};

interface IComponent {
  translationKey: string;
  $t: (s: string) => void;
}
export { translateKey };
