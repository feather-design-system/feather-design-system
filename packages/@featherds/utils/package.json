{
  "name": "@featherds/utils",
  "version": "0.12.31",
  "publishConfig": {
    "access": "public"
  },
  "author": "NantHealth",
  "scripts": {
    "compile": "tsc --declaration"
  },
  "keywords": [
    "vue",
    "FeatherUtils"
  ]
}
