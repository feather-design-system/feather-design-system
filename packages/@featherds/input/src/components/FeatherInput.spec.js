import { nextTick } from "vue";
import axe from "@featherds/utils/test/axe";
import FeatherInput from "./FeatherInput.vue";
import { mount } from "@vue/test-utils";
import * as id from "@featherds/utils/id";
import "@featherds/input-helper/test/MutationObserver";

jest.spyOn(id, "getSafeId").mockImplementation((x) => x);

const getWrapper = function (options) {
  document.body.innerHTML = "";
  options.attachTo = document.body;
  return mount(FeatherInput, options);
};

describe("FeatherInput.vue", () => {
  it("should render an input with a label", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input with a label and hint text", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        hint: "hint text",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input with a preset value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        hint: "hint text",
        modelValue: "YEET",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input with raised label when focused", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    await wrapper.find("input").trigger("focus");
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input with raised label when focused and has a value", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
    });
    await wrapper.find("input").trigger("focus");
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });

  it("should trigger input event when something is typed", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    const input = wrapper.find("input");
    const expected = "TEEST";
    input.element.value = expected;
    input.trigger("input");
    expect(wrapper.emitted("update:modelValue")[1][0]).toBe(expected);
  });

  it("should render an input error input", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input error input and has focus", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
      },
    });
    await wrapper.find("input").trigger("focus");
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render an input error input and has a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
        modelValue: "Test",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render a clear icon when clear has a value and input also has a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        clear: "clear",
        modelValue: "Test",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });

  it("should not render a clear icon when the input doesn't have a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        clear: "clear",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should render a maxlength attribute and character count when maxlength specified", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "test",
        maxlength: 5,
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  it("should clear input when clear-icon is clicked", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
        clear: "clear",
      },
    });
    wrapper.vm.handleClear();
    await nextTick();
    expect(wrapper.wrapperElement).toMatchSnapshot();
    expect(wrapper.emitted("update:modelValue")[1][0]).toBe("");
  });

  it("should allow custom focus events", async () => {
    const fn = jest.fn();
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
      attrs: {
        onFocus: fn,
      },
    });

    await wrapper.find("input").trigger("focus");
    expect(fn).toHaveBeenCalled();
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });

  it("should allow custom blur events", () => {
    const fn = jest.fn();
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
      attrs: {
        onBlur: fn,
      },
    });

    wrapper.find("input").element.focus();
    wrapper.find("input").element.blur();
    expect(fn).toHaveBeenCalled();
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
  describe("e2e elements, data-refs", () => {
    it("should have an input data-ref", () => {
      const wrapper = getWrapper({
        props: {
          label: "label",
          modelValue: "YEET",
        },
      });
      expect(wrapper.find("[data-ref-id='feather-input']").exists()).toBe(true);
    });

    it("should have a count data-ref", () => {
      const wrapper = getWrapper({
        props: {
          label: "label",
          modelValue: "YEET",
          hint: "HINT",
          maxlength: 5,
        },
      });
      expect(
        wrapper.find("[data-ref-id='feather-form-element-count']").exists()
      ).toBe(true);
    });
  });
  describe("a11y", () => {
    const accessibilityTest = async (options) => {
      const wrapper = mount(FeatherInput, options);
      expect(await axe(wrapper.element)).toHaveNoViolations();
    };
    it("should be accessible", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
        },
      });
    });
    it("should be accessible with no label", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
          hideLabel: true,
        },
      });
    });
    it("should be accessible with hint", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
          hint: "HINTTT",
        },
      });
    });
    it("should be accessible with error", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
          error: "Error",
        },
      });
    });
  });
});
