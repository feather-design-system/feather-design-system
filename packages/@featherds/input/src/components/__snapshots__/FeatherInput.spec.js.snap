// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatherInput.vue should allow custom blur events 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should allow custom focus events 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should clear input when clear-icon is clicked 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should not render a clear icon when the input doesn't have a value 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render a clear icon when clear has a value and input also has a value 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <a
          class="action-icon hide-when-disabled"
          data-ref-id="feather-form-element-clear"
          data-v-0de63cb2=""
          data-v-d1dfbdf8=""
          href="#"
          title="clear"
        >
          <svg
            aria-hidden="true"
            class="feather-icon"
            data-v-0aa06a12=""
            data-v-d1dfbdf8=""
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19,17.55,13.43,12,19,6.4A1,1,0,1,0,17.58,5L12,10.58,6.42,5A1,1,0,0,0,5,6.42L10.6,12,5,17.6A1,1,0,0,0,5,19a1,1,0,0,0,.71.29,1,1,0,0,0,.71-.3L12,13.41,17.6,19a1,1,0,0,0,.71.29A1,1,0,0,0,19,19,1,1,0,0,0,19,17.55Z"
            />
          </svg>
        </a>
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render a maxlength attribute and character count when maxlength specified 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="5"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    <div
      class="feather-input-count"
      data-ref-id="feather-form-element-count"
    >
      4 / 5
    </div>
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input error input 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container error feather-input-content error"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-ref-id="error-icon"
          data-v-0aa06a12=""
          data-v-0de63cb2=""
          data-v-49fd357e=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-51d97390=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input error input and has a value 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised error feather-input-content error"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-ref-id="error-icon"
          data-v-0aa06a12=""
          data-v-0de63cb2=""
          data-v-49fd357e=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-51d97390=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input error input and has focus 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised focused error feather-input-content error focused"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-ref-id="error-icon"
          data-v-0aa06a12=""
          data-v-0de63cb2=""
          data-v-49fd357e=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-51d97390=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input with a label 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input with a label and hint text 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <div
      class="feather-input-hint"
      data-ref-id="feather-form-element-hint"
      data-v-51d97390=""
    >
      hint text
    </div>
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input with a preset value 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <div
      class="feather-input-hint"
      data-ref-id="feather-form-element-hint"
      data-v-51d97390=""
    >
      hint text
    </div>
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input with raised label when focused 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherInput.vue should render an input with raised label when focused and has a value 1`] = `
<div
  class="feather-input-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-0de63cb2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-0de63cb2=""
    >
      <legend
        data-v-0de63cb2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-0de63cb2=""
      for="feather-input-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-0de63cb2=""
    >
      <div
        class="prefix"
        data-v-0de63cb2=""
      >
        
        
      </div>
      
      <input
        aria-describedby="feather-input-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-input"
        data-ref-id="feather-input"
        id="feather-input-label"
        maxlength="false"
        name="feather-input-label"
        type="text"
      />
      
      <div
        class="post"
        data-v-0de63cb2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-51d97390=""
    id="feather-input-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;
